src/hud.cpp

```diff
@@ -39,6 +43,81 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "touchscreengui.h"
 #endif
 
+#endif // SERVER
+
+MinimapModeMachine::MinimapModeMachine(u16 mode_count)
+{
+	assert(mode_count > 0); // Precondition
+	m_mode_count = mode_count;
+	m_defs = new MinimapModeDef[mode_count];
+}
+
+MinimapModeMachine::~MinimapModeMachine()
+{
+	delete[] m_defs;
+}
+
+void MinimapModeMachine::serialize(std::ostream &os)
+{
+	for (u16 i; i < m_mode_count; i++) {
```

Comment:
Another warning here: `warning: ‘i’ may be used uninitialized in this function [-Wmaybe-uninitialized]`